classy_vision/tasks/classification_task.py

```diff
@@ -707,9 +738,28 @@ def _set_model_train_mode(self):
         phase = self.phases[self.phase_idx]
         self.base_model.train(phase["train"])
 
+        if self.broadcast_buffers_before_eval and not self.train:
+            self._broadcast_buffers()
+
         if self.train and self.train_phase_idx >= 0:
             self.optimizer.update_schedule_on_epoch(self.where)
 
+    def _broadcast_buffers(self):
+        """Synchronize buffers across all devices.
+
+        This provides the eval time stability of `broadcast_buffers`,
+        especially for BatchNorm statistics which are not used during training,
+        without the latency of `broadcast_buffers`.
+        """
+        if self.distributed_model is None:
+            return
+        buffers = list(self.base_model.buffers())
+        if len(buffers) > 0:
+            logging.info("Synchronizing buffers before evaluation.")
+            _broadcast_coalesced(
```

Comment:
Note: this only handles the case where each GPU has a separate controlling process, which I believe is appropriate given Classy's process model, but I just wanted to flag it.

My understanding is that if a process controls multiple GPUs, we'd need a second intra-process synchronization across GPUs controlled by this process (as DistirbutedDataParallel's `_sync_params` method has).

We can avoid using an undocumented / private API here at the cost of some performance by broadcasting buffers individually. Since this is a once-per-epoch call, performance is not that important, so I'm open to either approach.